File: List3/Zadanie1C/PascalTriangleRow.cpp
#include <iostream>
#include "PascalTriangleRow.hpp"

int PascalTriangleRow::getBinom(int m){
    return row[m];
}

void PascalTriangleRow::solveRow(int m){
    for (int i=1;i<m;i++){
        row[i]=(row[i-1]*(m-i+1))/i;
    }
}

PascalTriangleRow::PascalTriangleRow(int n){
    row.assign(n+1, 1);
    solveRow(n);
}

File: List3/Zadanie1J/PascalTriangleRow.java
public class PascalTriangleRow {
    private int[] row;

    private void solveRow(int m){
        row[0]=1;
        row[m]=1;
        for (int i=1;i<m;i++){
            row[i]=(row[i-1]*(m-i+1))/i;
        }
    }

    public int getBinom(int m){
        return row[m];
    }

    public PascalTriangleRow(int n){
        row = new int[n+1];         
        solveRow(n);
    }
}
